src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettings.java

```diff
@@ -62,6 +102,12 @@ public SonarLintGlobalSettings getState() {
   @Override
   public void loadState(SonarLintGlobalSettings state) {
     XmlSerializerUtil.copyBean(state, this);
+    includedRules.forEach(it -> {
+      rules.put(it, new Rule(true));
+    });
+    excludedRules.forEach(it -> {
+      rules.put(it, new Rule(false));
+    });
```

Comment:
If you don't clear (or better set to null) the two sets after the migration, it means the next time you'll reopen your IDE, the migration will happen again, and override possible different values added in the meantime by the user in the rules map.